scRNA-seq/02-normalizing_scRNA-seq.Rmd

```diff
@@ -0,0 +1,255 @@
+installed already on your Docker container. 
+
+```{r setup}
+# Set seed for reproducibility
+set.seed(1234)
+
+# Magrittr for the pipe %>%
+library(magrittr)
+
+# GGPlot2 for the plots
+library(ggplot2)
+
+# Packages for single cell processing
+library(scater)
+library(scran)
+```
+
+Now let's set up the files we will be using:
+
+```{r files}
+# main data directory
+data_dir <- file.path("data", "glioblastoma")
+
+# Filtered count matrix file from previous notebook
+filtered_count_file <- file.path("data", "glioblastoma", "filtered_count_matrix.tsv")
+
+# Metadata file location
+metadata_file <- file.path(data_dir, "preprocessed", "darmanis_metadata.tsv")
+
+# Output directory
+norm_dir <- file.path(data_dir, "normalized")
+if (!dir.exists(norm_dir)) {
+  dir.create(norm_dir, recursive = TRUE)
+}
+```
+
+
+## Load the filtered count matrix and metadata
+
+```{r}
+sc_data_filtered <- readr::read_tsv(filtered_count_file)
+sc_metadata <- readr::read_tsv(metadata_file)
+```
+
+
+
+## Set up a SingleCellExperiment object from count data
+
+Now that we have filtered our data, we are ready to normalize it.
+To do this, we are going to use some single-cell RNA-seq R packages called `scater` and `scran`.
+
+![](diagrams/full-length_2.png)
+
+The functions in these packages require the data to be in their own special 
+object type (not an uncommon thing for R packages to do) called `SingleCellExperiment`.
+So first, we are going to set up our data in this format.
+To learn more about `SingleCellExperiment` objects and how it works with
+Bioconductor packages, we recommend this article: [Amezquita et al. _bioRxiv._ 2019.](https://www.biorxiv.org/content/10.1101/590562v1).
+
+```{r Set up SCE object}
+# Prep a matrix with just the rounded counts so we can create an sce object
+counts_mat <- round(as.matrix(sc_data_filtered[, -1]))
+
+```
+
+Put our counts matrix in this special `SingleCellExperiment` function: 
+
+```{r}
+# Set up the Single Cell Experiment object 
+sce_raw <- SingleCellExperiment::SingleCellExperiment(list(counts = counts_mat))
+
+# save some memory
+rm(counts_mat)
+
+# Store the gene names in this object
+rowData(sce_raw) <-  sc_data_filtered$gene
+```
+
+In the intro-to-R-tidyverse module notebook, `01-intro-to-r`, we discuss base R 
+object types, but there are some 'special' object types that are 
+package-specific. 
+`SingleCellExperiment` objects, for example, are what a lot of single-cell 
+analysis R packages use, so we will try to get acquainted with them. 
+
+Use this chunk we've set up below to do some exploring of the 
+`SingleCellExperiment` object you just made.
+
+```{r Explore SCE structure}
+# The SingleCellExperiment is a special type of object used by scater and scran
+# R packages, explore its properties here. 
+# Also check it out in the environment tab.
+```
+
+Below is a figure from [Amezquita et al. _bioRxiv._ 2019.](https://www.biorxiv.org/content/10.1101/590562v1) 
```

Comment:
Same comment about the citation.

Reply:
So, funny story, I just checked the published version, and this figure isn't in it. The new figure is arguably better, but way less compact. 
<img width="631" alt="Screen Shot 2020-07-13 at 2 19 56 PM" src="https://user-images.githubusercontent.com/20483/87338977-eed1d180-c513-11ea-911f-8fac52393e9a.png">

Pushing this decision to the next PR, but noting it here.